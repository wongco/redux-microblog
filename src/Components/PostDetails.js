import React, { Component } from 'react';
import styled from 'styled-components';
import Vote from '../Components/Vote';

const StyledPostDetail = styled.div`
  width: 50vw;
`;

const StyledTop = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0;
`;

const StyledHeader = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 0;
  margin: 0;
`;

const StyledHeader2 = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 0;
`;
class PostDetails extends Component {
  render() {
    const { title, description, body, votes } = this.props.post;
    return (
      <StyledPostDetail>
        <StyledTop>
          <StyledHeader>
            <h1>{title}</h1>
            <div>
              <button onClick={this.props.toggleEdit}>Edit Post</button>
              <button onClick={this.props.deletePost}>Delete Post</button>
            </div>
          </StyledHeader>
          <StyledHeader2>
            <div>{description}</div>
            <Vote votes={votes} voteAction={this.props.voteAction} />
          </StyledHeader2>
        </StyledTop>
        <div>{body}</div>
      </StyledPostDetail>
    );
  }
}

PostDetails.defaultProps = {
  post: {
    title: 'Amazing things',
    description: 'an excercise in biting the dust',
    body: 'eating all day all night lorem ipsum another one bites the dust',
    votes: 1
  },
  toggleEdit: () => console.log('toggle editing in PostView'),
  deletePost: () => console.log('delete the post from redux state'),
  voteAction: () => console.log('voting in PostView ')
};

export default PostDetails;
