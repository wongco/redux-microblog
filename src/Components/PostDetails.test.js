import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostDetails from './PostDetails';

it('renders without crashing', () => {
  shallow(<PostDetails />);
});

it('matches snapshot for default criteria', () => {
  const wrapper = shallow(<PostDetails />);
  const serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
