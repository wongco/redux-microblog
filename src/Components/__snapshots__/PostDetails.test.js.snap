// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for default criteria 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <h1>
        Amazing things
      </h1>
      <styled.div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f044",
                "M402.3 344.9l32-32c5-5 13.7-1.5 13.7 5.7V464c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h273.5c7.1 0 10.7 8.6 5.7 13.7l-32 32c-1.5 1.5-3.5 2.3-5.7 2.3H48v352h352V350.5c0-2.1.8-4.1 2.3-5.6zm156.6-201.8L296.3 405.7l-90.4 10c-26.2 2.9-48.5-19.2-45.6-45.6l10-90.4L432.9 17.1c22.9-22.9 59.9-22.9 82.7 0l43.2 43.2c22.9 22.9 22.9 60 .1 82.8zM460.1 174L402 115.9 216.2 301.8l-7.3 65.3 65.3-7.3L460.1 174zm64.8-79.7l-43.2-43.2c-4.1-4.1-10.8-4.1-14.8 0L436 82l58.1 58.1 30.9-30.9c4-4.2 4-10.8-.1-14.9z",
              ],
              "iconName": "edit",
              "prefix": "far",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          style={
            Object {
              "color": "orange",
              "cursor": "pointer",
            }
          }
          symbol={false}
          title=""
          transform={null}
        />
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f2ed",
                "M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z",
              ],
              "iconName": "trash-alt",
              "prefix": "far",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          style={
            Object {
              "color": "red",
              "cursor": "pointer",
            }
          }
          symbol={false}
          title=""
          transform={null}
        />
      </styled.div>
    </styled.div>
    <styled.div>
      <h3>
        an excercise in biting the dust
      </h3>
      <Vote
        voteAction={[Function]}
        votes={1}
      />
    </styled.div>
  </styled.div>
  <div>
    eating all day all night lorem ipsum another one bites the dust
  </div>
</styled.div>
`;
